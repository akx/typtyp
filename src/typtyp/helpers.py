from typing import Hashable, Iterable, TypeVar

THashable = TypeVar("THashable", bound=Hashable)


def unique_in_order(values: Iterable[THashable]) -> Iterable[THashable]:
    seen = set()
    for v in values:
        if v in seen:
            continue
        seen.add(v)
        yield v
