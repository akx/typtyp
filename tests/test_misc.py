from typing import TypedDict

from tests.helpers import check_with_tsc, typescript_from_types


class DoublyNumber(TypedDict):
    # Will end up `number | string`, not `number | string | number`
    value: float | str | int


def test_duplicate_union_member(snapshot):
    """
    Test that union members with types that end up being the same in TypeScript
    (e.g., float and int both become number) are handled correctly.
    """
    code = typescript_from_types([DoublyNumber])
    assert check_with_tsc(code)
    assert code == snapshot
