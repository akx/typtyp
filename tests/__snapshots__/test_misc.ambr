# serializer version: 1
# name: test_duplicate_union_member
  '''
  export interface DoublyNumber {
  value: number | string
  }
  
  '''
# ---
